### DISCUSSION TITLE
Additional attributes to standard nodes

### DISCUSSION THREAD

**Discussion by asaf** [markdoc/markdoc#331] [Q&A]
Hey,

I would like to annotate my headers with an extra attribute (e.g., `data-file=foo.js`) to the standard nodes, e.g.,:

```md
## my title {% data-file=foo.js %/}
```

I tried adding a new attribute to the `heading` node but it doesn't propagate the value to the `transform` function.

I don't want to replace the standard headings format as it breaks our markdowns standards,

Is there any way to do this?

Best,
Asaf.

**Comment by asaf** [comment]
I could make this work by wrapping the value in quotes as explained [here](https://github.com/markdoc/markdoc/discussions/318#discussioncomment-4821979)
(a bit confusing because id and classes do not need quotes)

**Comment by segphault** [comment]
Yes, the attribute values are written like JSON, so the string does need to be quoted. In addition to quoting the string, you also need to remove the forward-slash. The slash is only needed for named, self-closing tags (and goes before the percent sign rather than after). This is the syntax that you're looking for:

```
## my title {% data-file="foo.js" %}
```

When written this way, the attribute will appear in `node.attributes['data-file']` in the `transform` function. In order to get it to validate (and pass through into the render tree without writing a transform function) you also have to override the heading node in the schema: 

```
const config = {
  nodes: {
    heading: {
      ...Markdoc.nodes.heading,
      attributes: {
        ...Markdoc.nodes.heading.attributes,
        'data-file': { type: String }
      }
    }
  }
};
```
